web/source/dom/domEventHandlers.ts

```diff
@@ -734,6 +675,156 @@ namespace com.keyman.dom {
       }
     }
 
+    private performCaretSearch(target: TouchAliasElement, scroller: HTMLElement, touchX: number, touchY: number) {
```

Comment:
This seems to have changed somewhat from the previous code. I have done a comparison and think I can see the differences okay.

Also, please add a code comment describing the function.